plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/ExpressionConverter.java

```diff
@@ -115,94 +111,97 @@ private static Expression toIcebergExpression(String columnName, Type type, Doma
-                    }
-                    else { // case between
-                        Expression between = and(
-                                greaterThanOrEqual(columnName, getIcebergLiteralValue(type, low)),
-                                lessThanOrEqual(columnName, getIcebergLiteralValue(type, high)));
-                        expression = or(expression, between);
-                    }
-                }
-                else {
-                    if (lowBound == EXACTLY && low.getValueBlock().isPresent()) {
-                        // case >=
-                        expression = or(expression, greaterThanOrEqual(columnName, getIcebergLiteralValue(type, low)));
-                    }
-                    else if (lowBound == ABOVE && low.getValueBlock().isPresent()) {
-                        // case >
-                        expression = or(expression, greaterThan(columnName, getIcebergLiteralValue(type, low)));
-                    }
-
-                    if (highBound == EXACTLY && high.getValueBlock().isPresent()) {
-                        // case <=
-                        if (low.getValueBlock().isPresent()) {
-                            expression = and(expression, lessThanOrEqual(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                        else {
-                            expression = or(expression, lessThanOrEqual(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                    }
-                    else if (highBound == BELOW && high.getValueBlock().isPresent()) {
-                        // case <
-                        if (low.getValueBlock().isPresent()) {
-                            expression = and(expression, lessThan(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                        else {
-                            expression = or(expression, lessThan(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                    }
-                }
+                expression = or(expression, toIcebergExpression(columnName, range));
             }
             return expression;
         }
 
         throw new VerifyException("Did not expect a domain value set other than SortedRangeSet but got " + domainValues.getClass().getSimpleName());
     }
 
-    private static Object getIcebergLiteralValue(Type type, Marker marker)
+    private static Expression toIcebergExpression(String columnName, Range range)
+    {
+        Type type = range.getType();
+
+        if (range.isLowInclusive() && range.isHighInclusive()) {
+            Object icebergLow = getIcebergLiteralValue(type, range.getLowBoundedValue());
+            Object icebergHigh = getIcebergLiteralValue(type, range.getHighBoundedValue());
+
+            // Note that this can be different from range.isSingleValue() condition if getIcebergLiteralValue rounds values. TODO verify if this can be the case
```

Comment:
> if getIcebergLiteralValue rounds values

IIUC, this would be a bigger correctness concern rather than just the `isSingleValue` case, right? so might be better to add this TODO on `getIcebergLiteralValue` itself - may be even the suspicious type condition.

Reply:
I read that code a little bit more closely and i think this is not an issue, will remove TODO & simplify.